### PR TITLE
Add Replay mode

### PR DESCRIPTION
**Pull request by tysmith**
I plan to merge this tomorrow morning. Have a look and let me know if there are any blockers. Feel free to open issues for any missing features.

### CODE REVIEW

File: grizzly/reduce/reduce.py [hunk 1]
```diff
@@ -661,95 +633,72 @@ def _dyn_resp_close():
...
-            self._target.check_relaunch()
+        elif runner.result == Runner.IGNORED:
+            LOG.info("Uninteresting: ignored")
+            self._target.close()
 
-        finally:
-            iteration_done_event.set()
-            if self._idle_poll:
-                poll.join()
+            # TODO: why is this here?
```
Comment (jschwartzentruber) [inline grizzly/reduce/reduce.py:692]: For no good reason. Delete it.

File: grizzly/reduce/test_common.py [hunk 2]
```diff
@@ -145,6 +148,7 @@ def lithium_init(self):
         pass
 
     def _get_location(self):
+        # TODO: is this dead code?
```
Comment (jschwartzentruber) [inline grizzly/reduce/test_common.py:151]: Yes. Delete it.

File: grizzly/__main__.py [hunk 3]
```diff
@@ -2,12 +2,17 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
-
 import sys
 
-from .main import console_main
+import grizzly.adapters
```
Comment (jschwartzentruber) [inline grizzly/__main__.py:7]: This should be relative.

File: grizzly/reduce/reduce.py (hunk 1 above)
Comment (tysmith) [reply grizzly/reduce/reduce.py]: Done. 19b46946

File: grizzly/reduce/test_common.py (hunk 2 above)
Comment (tysmith) [reply grizzly/reduce/test_common.py]: Done. b900148

File: grizzly/__main__.py (hunk 3 above)
Comment (tysmith) [reply grizzly/__main__.py]: Done. c1c30cd

**Pull request merged**